21 upgrades erstellen

fcavar opened the pull request:
Upgrades hinzugefügt und teilweise auswählbar (siehe letzten Commit). Es fehlt defintiv noch das gesamte Upgrade Auswahl Menu Management, bevor man hier weiter die Upgrades ausbaut. Sobald dieses steht, kann man hier balancen, wie viele und wie stark die Upgrades sein sollen.

fcavar committed:
Merge pull request #29 from elloco02/develop

Diesen branch up to date bringen mit develop.

fcavar committed:
Upgrades hinzugefügt

Für Player:
- HP hinzugefügt Upgrade (noch nicht auswählbar, weil HP Management fehlt)
- HP Regeneration Upgrade hinzugefügt (noch nicht auswählbar, weil HP Regeneration Management fehlt)
- Move Speed Upgrade hinzugefügt

Für Weapon:
- alle 3 Upgrades hinzugefügt und auswählbar

Momentan immernoch temporäre Lösung durch "ESC" das Upgradefenster anzuzeigen. Deshalb immernoch verbuggt, dass sich die Upgrades mehrfach stacken, wenn man mehrfach "ESC" drückt und "Resume Game"

Tonka-de commented on neues-spiel/player/upgrades/health_points_upgrade.gd:
wont work since the player itself does not have a "max_health". Just the Health-Component has that. And when adding to max health, then the player should most likely also be healed by the same amount...

Tonka-de commented on neues-spiel/player/upgrades/regeneration_upgrade.gd:
not an attribute in player once again, most likely part of the Health-Component

Tonka-de commented on neues-spiel/weapon/gun.gd:
```diff
@@ -1,27 +1,29 @@
 	refill_ammo()
+	# loop over all bullet upgrades and apply
+	for upgrade in bullet_upgrades:
+		upgrade.apply_upgrade(weapon_attack)
```
we most likely want to give the actual weapon, something like firerate is not part of the actual "attack"

Tonka-de replied:
except if it will be the case that is, but currently does not seem like it

fcavar committed:
HP und Regenerationsupgrade angepasst für health_component

- health points upgrade wird auf health_component angewandt
- regeneration upgrade wird ebenfalls auf health_component angewandt, indem die health_regenration_time reduziert wird

fcavar committed:
weapon_attack für gun.gd angepasst

- Verlagerung aller Stats in die weapon_attack.gd
- Anpassung von gun.gd

fcavar committed:
Merge branch 'develop' into 21-upgrades-erstellen
